spackbot/pr_add_reviewers.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import contextlib
+import logging
+import os
+import re
+import tempfile
+
+import gidgethub
+from sh.contrib import git
+from gidgethub import routing
+
+logger = logging.getLogger(__name__)
+router = routing.Router()
+
+package_path = r"^var/spack/repos/builtin/packages/(\w[\w-]*)/package.py$"
+
+non_reviewers_comment = """\
+  @{non_reviewers} can you review this PR?
+
+  This PR modifies the following package(s), for which you are listed as a maintainer:
+
+  * {packages_with_maintainers}
+"""
+
+no_maintainers_comment = """\
+Hi @{author}! I noticed that the following package(s) don't yet have maintainers:
+
+* {packages_without_maintainers}
+
+Are you interested in adopting any of these package(s)? If so, simply add the following to the package class:
+```python
+    maintainers = ['{author}']
+```
+If not, could you contact the developers of this package and see if they are interested? Please don't add maintainers without their consent.
+
+_You don't have to be a Spack expert or package developer in order to be a "maintainer", it just gives us a list of users willing to review PRs or debug issues relating to this package. A package can have multiple maintainers; just add a list of GitHub handles of anyone who wants to volunteer._
+"""
+
+
+async def changed_packages(gh, pull_request):
```

Comment:
Someday we may want to refactor things and move this to a separate file so multiple bot functions can use it, but let's way until we actually need to. "Premature optimization is the root of all evil" is the only thing I've learned in grad school...